testing/E2E_Test_Update.md

```diff
@@ -0,0 +1,164 @@
+suite of tests in a single repository that contains smart abstractions can be major factors 
+in the reduction of the amount of code that will be written and in turn that needs to be 
+maintained.  Common design of application means that only a single test needs to be written 
+to cover all platforms that need to be covered if abstractions such as page objects and 
+data objects are used to remove selectors and expected data from the the tests themselves.  
+These abstractions can then be passed into the tests for multiple platforms.  Indeed, the 
+Java-client has support for cross-platform page objects through the use of special 
+annotations.
+
+## Updated Proposals
+
+A decision should first be made if preferred selectors are to be implmented in the mobile
+showcase applications.  The effects of this decision are shown below:
+
+#### 1). Using Preferred Selectors in Applications
+
+* Tests are likely to be less brittle, therefore less time should be required on 
+maintenance
+* Any client can be used provided it has the ability to handle redirects to the browser
+
+
+#### 2). Not Using Preferred Selectors in Applications
+
+* Tests are likely to be more brittle, therefore more time required to maintain them
+* The Java-client should be the preferred choice of client as it has more strategies to
+target locators than other libraries
+
+### Possible Test Suite Implementations
+
+1. A nodejs implmentation with webdriverio - This should be considered if there is a 
+decision to use preferred selecotrs is to use them in the showcase applications as there 
+is a lot of experience using nodejs within the teams.
+
+1. A Java-client implementation or the Akow library - should be considered if the decision
+on using preferred selectors is not to use them. This will allow for more locator strategies 
+that can be used to create the tests.  The downside of this approach could be the upkeep of 
+the Akow library. 
+
+## Test Writing Time Estimation
+
+### Writing New tests
+
+A new test should take approximately 3-4 hours to complete regardless of the language of 
+implementation.  Appium Inspector can be used to find the selectors and also generate an 
+initial version of the code that is unrefined.  The person developing the test will need 
+to extract the selectors and generate page objects and test case files based on the 
+unrefined code.  
+
+Through the use of best practices such as page objects, areas of a test that have been 
+previously covered by code inside the page object can be quickly reused in the new test 
+cases and the only selectors and code to be created will belong to pages that have yet 
+to be covered by a test case.
+
+However, in cases where preferred selector strategies cannot be used or in cases where 
+there may be some problems with the application some extra time may be required.
+
+
+### Maintaining tests
+
+The majority of time spent to make changes to maintain the test suite should be minimal 
+(1 – 2 hours).  
```

Comment:
> (1 – 2 hours)
daily? weekly? monthly?

Reply:
I think it was meant per week, but I think it's an useless information, because it depends on how many changes were done.

Reply:
All times are per test. This includes time for code review and testing by others too. If implemented well then times could be much shorter.